pkg/api/resource/quantity.go

```diff
@@ -0,0 +1,422 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resource
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"math/big"
+	"regexp"
+	"strings"
+
+	"speter.net/go/exp/math/dec/inf"
+)
+
+// Quantity is a fixed-point representation of a number.
+// It provides convenient marshaling/unmarshaling in JSON and YAML,
+// in addition to String() and Int64() accessors.
+//
+// The serialization format is:
```

Comment:
I feel this BNF can be clearer

```
<serialized>      ::= <sign><value> | <value>
<sign>            ::= "+" | "-"
<value>           ::= <number><exponent> | <number>
<number>          ::= <digits> | <digits>.<digits>
<digits>          ::= <digit> | <digit><digits>
<digit>           ::= 0 | 1 | ... | 9
<exponent>        ::= <binarySuffix> | <decimalExponent> | <decimalSuffix>
<binarySuffix>    ::= i | Ki | Mi | Gi | Ti | Pi | Ei
<decimalSuffix>   ::= m | k | M | G | T | P | E
<decimalExponent> ::= "e" <digits> | "E" <digits>
```

or, if you buy my other argument about E exponents being a way to write a number :

```
<serialized>      ::= <sign><value> | <value>
<sign>            ::= "+" | "-"
<value>           ::= <number><suffix> | <number>
<number>          ::= <int> | <float>
<int>             ::= <digits> | <digits><exponent>
<float>           ::= <bareFloat> | <bareFloat><exponent>
<bareFloat>       ::= <digits>.<digits> | .<digits> | <digits>.
<digits>          ::= <digit> | <digit><digits>
<digit>           ::= 0 | 1 | ... | 9
<suffix>          ::= <binarySuffix> | <decimalSuffix>
<binarySuffix>    ::= i | Ki | Mi | Gi | Ti | Pi | Ei
<decimalSuffix>   ::= m | k | M | G | T | P | E
<exponent>        ::= "e" <digits> | "E" <digits>
```